src/Utils.hs

```diff
@@ -250,3 +253,41 @@ runLog ::
 runLog logger =
   interpret \case
     Output o -> embed $ logger o
+
+envToken :: IO (Maybe Text)
+envToken = fmap tshow <$> getEnv "GITHUB_TOKEN"
+
+localToken :: IO (Maybe Text)
+localToken = do
+  exists <- fileExist "github_token.txt"
+  if exists
+    then (Just . T.strip <$> T.readFile "github_token.txt")
+    else (return Nothing)
+
+hubFileLocation :: IO (Maybe FilePath)
+hubFileLocation = do
+  xloc <- fmap (<> "/hub") <$> getEnv "XDG_CONFIG_HOME"
+  hloc <- fmap (<> "/.config/hub") <$> getEnv "HOME"
+  return (xloc <|> hloc)
+
+hubConfigToken :: IO (Maybe Text)
+hubConfigToken = do
+  hubFile <- hubFileLocation
+  case hubFile of
+    Nothing -> return Nothing
+    Just file -> do
+      exists <- fileExist file
+      if not exists
+        then return Nothing
+        else do
+          contents <- T.readFile file
+          let splits = T.splitOn "oauth_token: " contents
+              token = T.takeWhile (/= '\n') $ head (drop 1 splits)
+          return $ Just token
+
+getGithubToken :: IO Text
+getGithubToken = do
+  et <- envToken
+  lt <- localToken
+  ht <- hubConfigToken
+  return $ fromJust (et <|> lt <|> ht)
```

Comment:
Looks nice!  I might deco the localToken just to keep the code simple, but it's probably a good idea to leave it for at least 1 deployment while you shift the installation.